[4924] Switch 3.1.x sites to run within Next UI

aachavarria opened the pull request:
https://github.com/craftercms/craftercms/issues/4924
https://github.com/craftercms/craftercms/issues/4979

aachavarria committed:
Creating LegacyComponentsPanel.tsx

rart commented on ui/app/src/components/LegacyComponentsPanel/LegacyComponentsPanel.tsx:
```diff
@@ -0,0 +1,125 @@
+          translation: {
+            addComponent: 'Add Component',
+            components: 'Components',
+            done: 'done'
```
translate

aachavarria committed:
creating useEditMode.ts hook;
adding formatMessage
reading getStoredLegacyComponentPanel
opening and closing panel

aachavarria committed:
fixing issue on lookupsitefolder dont sending token

aachavarria committed:
creating BrowseFilesDialog.tsx
creating FolderBrowserTreeView.tsx
updating MediaCard.tsx to support selection

rart commented on ui/app/src/components/BrowseFilesDialog/BrowseFilesDialog.tsx:
```diff
@@ -0,0 +1,74 @@
+import { BrowseFilesDialogContainer } from './BrowseFilesDialogContainer';
+import StandardAction from '../../models/StandardAction';
+
+interface BrowseFilesDialogBaseProps {
```
```suggestion
export interface BrowseFilesDialogBaseProps {
```

rart commented on ui/app/src/components/BrowseFilesDialog/BrowseFilesDialogContainer.tsx:
```diff
@@ -0,0 +1,238 @@
+import FolderBrowserTreeView from '../Navigation';
+import Box from '@material-ui/core/Box';
+
+interface BrowseFilesDialogUIProps {
```
```suggestion
export interface BrowseFilesDialogUIProps {
```

rart commented on ui/app/src/components/Navigation/FolderBrowserTreeView.tsx:
```diff
@@ -14,271 +14,131 @@
+    if (currentPath) {
+      let nodesLookup = nodesLookupRef.current;
+      if (nodesLookup[currentPath] && nodesLookup[currentPath]?.fetched) {
+      } else {
```
```suggestion
      if (nodesLookup[currentPath] && nodesLookup[currentPath]?.fetched) {
        // either add something here, or flip the condition
      } else {
```

rart commented on ui/app/src/components/Navigation/FolderBrowserTreeView.tsx:
```diff
@@ -14,271 +14,131 @@
+          requests.push(
+            get(
+              `/studio/api/1/services/api/1/content/get-items-tree.json?site=${site}&path=${nextPath}&depth=1&order=default`
+            )
```
I think we have a service for this endpoint already.

aachavarria committed:
creating FolderBrowserTreeViewSkeleton.tsx

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/4924

aachavarria committed:
updating FolderBrowserTreeView.tsx to receive onIconClick and onLabelClick
adding BrowseDialog to LegacyComponentsPanel.tsx

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/4924

aachavarria committed:
using fetchLegacyItemsTree on FolderBrowserTreeView.tsx

aachavarria committed:
adding legacyContentModel on LegacyComponentsPanel.tsx
creating legacy utils to parse xml;

aachavarria committed:
adding LegacyLoadFormDefinition

aachavarria committed:
creating onComponentDrop

aachavarria committed:
using insertInstance to insert component from browseFileDialog

aachavarria committed:
adding insert and sort operations to LegacyComponentsPanel.tsx

aachavarria committed:
adding datasource to insertInstance

aachavarria committed:
adding deleteItem

aachavarria committed:
adding comment

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/4924

aachavarria committed:
Testing and Fixes with nestedDropzones

aachavarria committed:
adding isInsert to componentDropped
adding support to move on LegacyComponentsPanel.tsx

aachavarria committed:
adding newEmbedded to LegacyFormDialog.tsx to support the creation of embedded component using the form engine

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/4924

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/4924

aachavarria committed:
forms-engine.js: adding action and callback to OPEN_CHILD_COMPONENT;
adding ability to create embedded components to embedded_legacy_form.ftl;

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/4924

aachavarria committed:
adding styles to browse components button on components panel

aachavarria committed:
creating DRAG_AND_DROP_COMPONENTS_PANEL_CLOSED on dnd-controller.js;
adding dndOn to guest.js
listening DRAG_AND_DROP_COMPONENTS_PANEL_CLOSED on LegacyComponentsPanel.tsx;

aachavarria committed:
fixing issues and cleaning code

aachavarria committed:
creating util fetchAndInsertContentInstance

aachavarria committed:
improvements

aachavarria committed:
fixing issue with the selected index on node-selector.js;
adding index to embedded-content.js shared-content.js dropTargets.js

rart commented on templates/web/legacy/embedded_legacy_form.ftl:
```diff
@@ -98,6 +99,8 @@
         var isHidden = CStudioAuthoring.Utils.getQueryVariable(location.search, 'isHidden') === 'true';
         var changeTemplate = CStudioAuthoring.Utils.getQueryVariable(location.search, 'changeTemplate');
 
+        const embeddedData = newEmbedded?  JSON.parse(newEmbedded) : false;
```
```suggestion
        const embeddedData = newEmbedded ? JSON.parse(newEmbedded) : false;
```

rart commented on templates/web/legacy/embedded_legacy_form.ftl:
```diff
@@ -146,19 +149,24 @@
                                     CStudioAuthoring.InContextEdit.messageDialogs({
                                       type: 'OPEN_CHILD_COMPONENT',
-                                      key: modelId,
+                                      key: Boolean(modelId)? modelId : null,
```
```suggestion
                                      key: Boolean(modelId) ? modelId : null,
```

rart commented on templates/web/legacy/embedded_legacy_form.ftl:
```diff
@@ -146,19 +149,24 @@
-                                  if (!modelId) {
-                                    window.top.postMessage({ type: 'EMBEDDED_LEGACY_FORM_RENDERED' }, '*');
-                                  } else {
+                                  // TODO: embeddedData, support create embedded component
```
```suggestion
```

rart commented on templates/web/legacy/embedded_legacy_form.ftl:
```diff
@@ -146,19 +149,24 @@
+                                      edit: Boolean(modelId),
+                                      selectorId: embeddedData? embeddedData.fieldId : null,
+                                      ds: embeddedData? embeddedData.datasource : null,
+                                      order: embeddedData? embeddedData.index : null,
```
```suggestion
                                      contentType: embeddedData ? embeddedData.contentType : null,
                                      edit: Boolean(modelId),
                                      selectorId: embeddedData ? embeddedData.fieldId : null,
                                      ds: embeddedData ? embeddedData.datasource : null,
                                      order: embeddedData ? embeddedData.index : null,
```

rart commented on ui/app/src/components/BrowseFilesDialog/BrowseFilesDialog.tsx:
```diff
@@ -0,0 +1,75 @@
+  numOfLoaderItems?: number;
+}
+
+export type BrowseFilesDialogProps = PropsWithChildren<
```
Should extend DialogProps

rart commented on ui/app/src/components/BrowseFilesDialog/BrowseFilesDialogContainer.tsx:
```diff
@@ -0,0 +1,239 @@
+    setCurrentPath(path);
+  };
+
+  return (
```
Split out UI

rart commented on ui/app/src/components/LegacyComponentsPanel/LegacyComponentsPanel.tsx:
```diff
@@ -0,0 +1,613 @@
+  path: string;
+  isNew: boolean | 'existing';
+  trackingNumber: string;
+  zones: LookupTable<Array<any>>;
```
Add type to array items

rart commented on ui/app/src/components/Navigation/FolderBrowserTreeViewUI.tsx:
```diff
@@ -0,0 +1,314 @@
+  onKeyPress?(event: React.KeyboardEvent): void;
+}
+
+export default function FolderBrowserTreeViewUI(props: FolderBrowserTreeViewUIProps) {
```
Split components into different files

rart commented on ui/app/src/modules/Preview/PreviewConcierge.tsx:
```diff
@@ -117,7 +117,7 @@ import { UNDEFINED } from '../../utils/constants';
 import { useSiteUIConfig } from '../../utils/hooks/useSiteUIConfig';
 
-const guestMessages = defineMessages({
+export const guestMessages = defineMessages({
```
move to assets/guestMessages

rart commented on ui/app/src/services/content.ts:
```diff
@@ -425,10 +426,16 @@ export function deleteItem(
       .remove();
+
+    if ($fieldNode.children().length === 0) {
+      $fieldNode.html('');
```
```suggestion
      // If the node isn't completely blank, the xml formatter won't do it's job in converting to a self-closing tag.
      // Also, later on, when retrieved, some *legacy* functions would impaired as the deserializing into JSON had unexpected content
      $fieldNode.html('');
```

rart reviewed (changes requested):
* For components/Navigation/FolderBrowserTreeView*, move to components/FolderBrowserTreeView/*

rart committed:
Update templates/web/legacy/embedded_legacy_form.ftl

rart committed:
Update templates/web/legacy/embedded_legacy_form.ftl

rart committed:
Update templates/web/legacy/embedded_legacy_form.ftl

aachavarria committed:
embedded_legacy_form.ftl: removing comment;
exporting guestMessages.ts to assets;
adding DialogProps to BrowseFilesDialog.tsx;
creating BrowseFilesDialogUI.tsx;
splitting out FolderBrowserTreeView.tsx
